### ISSUE TITLE
id values are not unique in _walk_refs

### ISSUE THREAD

**Issue by geographika** [gazpachoking/jsonref#71]
The wrong JSON schema is inserted in a ref. Test case attached. Only after debugging did I realise this is a duplicate of #61. 
To recreate extract [test_case.zip](https://github.com/user-attachments/files/17652806/test_case.zip) and run test.py (shown below). 

Test on a Windows machine:

```
cd test_case
C:\Python310\Scripts\pip install jsonref
C:\Python310\python test.py
```
```python
import jsonref
import json
from pathlib import Path

# Get the CWD as a file URI
cwd_uri = Path.cwd().as_uri() + "/"

with open("map.json", encoding="utf-8") as f:
    jsn_schema = jsonref.load(
        f, base_uri=cwd_uri, lazy_load=False, proxies=False
    )

# print(json.dumps(jsn_schema, indent=4))

print(json.dumps(jsn_schema["properties"]["outputformats"]["items"]["properties"]["transparent"], indent=4))

# incorrect output below - expression.json is in the place of onoff.json
"""
{
    "oneOf": [
        {
            "type": "boolean"
        },
        {
            "oneOf": [
                {
                    "type": "array",
                    "items": {
                        "type": "number",
                        "minimum": -1,
                        "maximum": 255
                    },
                    "minItems": 3,
                    "maxItems": 3
                },
                {
                    "pattern": "^#([a-fA-F0-9]{6,8}|[a-fA-F0-9]{3,4})$",
                    "example": "#aa33cc",
                    "type": "string"
                }
            ]
        }
    ]
}
"""

# remove part of the JSON e.g. "labels" in legend.json and the correct JSON is returned

"""
{
    "oneOf": [
        {
            "type": "boolean"
        },
        {
            "type": "string",
            "enum": [
                "on",
                "off"
            ]
        }
    ]
}
"""
```

**Comment by geographika** [comment]
Fixed with e827f23 - test case in this ticket now produces the correct output, both with `lazy_load=True, proxies=True` and `lazy_load=False, proxies=False`.

Thanks @gazpachoking !

```
pip install git+https://github.com/gazpachoking/jsonref.git@master
python test.py
```

Output:

```json
{
    "oneOf": [
        {
            "type": "boolean"
        },
        {
            "type": "string",
            "enum": [
                "on",
                "off"
            ]
        }
    ]
}
```
